### ISSUE TITLE
PIC / ES Freezes for minutes at a time with no internet.

### ISSUE THREAD

**Issue by AmandaCameron** [PowerShell/vscode-powershell#1446]
## Issue Description ##

When using the extension on my laptop when I have no internet, PIC and the Editor Services are both super slow. It took a good 5-minutes to scaffold out a project using plaster, and the intellisense never seems to load.

If I had to guess it's caused by something making a network call that takes too long and eventually times out.

## Attached Logs ##

[vscode-logs.zip](https://github.com/PowerShell/vscode-powershell/files/2216994/vscode-logs.zip)

### Visual Studio Code ###

| Name | Version |
| --- | --- |
| Operating System | Linux x64 4.15.0-29-generic |
| VSCode | 1.25.1|
| PowerShell Extension Version | 1.8.1 |

### PowerShell Information ###

|Name|Value|
|---|---|
|PSVersion|6.1.0-preview.4|
|PSEdition|Core|
|GitCommitId|6.1.0-preview.4|
|OS|Linux 4.15.0-29-generic #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018|
|Platform|Unix|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.10032.0 6.1.0-preview.4|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|
|WSManStackVersion|3.0|

### Visual Studio Code Extensions ###

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|bazel-code|DevonDCarew|0.1.9|
|better-toml|bungcip|0.3.2|
|clang-format|xaver|1.6.2|
|cpptools|ms-vscode|0.17.7|
|csharp|ms-vscode|1.15.2|
|debugger-for-chrome|msjsdiag|4.8.0|
|EditorConfig|EditorConfig|0.12.4|
|gitlab-workflow|fatihacet|0.6.0|
|gitlens|eamodio|8.5.2|
|Go|ms-vscode|0.6.84|
|HCL|wholroyd|0.0.5|
|jinja|wholroyd|0.0.8|
|Lisp|mattn|0.1.8|
|markdown-table-formatter|fcrespo82|1.3.2|
|much-assembly-required-language-support|PJB3005|0.1.5|
|much-assembly-required-upload-on-save|tomhodder|0.0.6|
|opa|tsandall|0.0.6|
|polymer-ide|polymer|0.6.0|
|poptheme|ArtisanByteCrafter|0.0.8|
|postgresql|JPTarquino|0.0.2|
|PowerShell|ms-vscode|1.8.1|
|project-manager|alefragnani|8.0.0|
|project-snippets|rebornix|0.5.0|
|terraform|mauve|1.2.3|
|vs-remote-debugger|mkloubert|2.0.0|
|vscode-dash|deerawan|1.10.0|
|vscode-docker|PeterJausovec|0.0.27|
|vscode-eslint|dbaeumer|1.4.12|
|vscode-helm|technosophos|0.3.0|
|vscode-kubernetes-tools|ms-kubernetes-tools|0.1.11|
|vscode-lua|trixnz|0.12.4|
|vscode-proto3|zxh404|0.2.1|
|vscode-yaml|redhat|0.0.13|
|vsliveshare|ms-vsliveshare|0.3.423|
|xml|DotJoshJohnson|2.3.1|
|zenscript|Indexyz|0.0.2|;

</details>

**Comment by rjmholt** [comment]
Hi @AmandaCameron, thanks for opening an issue! As of v1.8.0 we don't do any networking at all in the PowerShell extension (in fact, we don't support TCP connections any more at all -- see https://github.com/PowerShell/vscode-powershell/issues/954).

It may be that VSCode itself trying to do a lot of networking -- see https://github.com/Microsoft/vscode/issues/54354.

**Comment by rkeithhill** [comment]
Another option might be PowerShell trying to verify the Authenticode signature on Start-EditorServices.ps1.  We've observed this behavior before with some code we have that validates assemblies are signed with a valid Authenticode cert.  It is usually something like a three minute hang/delay IIRC.  Although this was usually for systems that had never been connected to the internet (DOD stuff).

**Comment by rjmholt** [comment]
Good call @rkeithhill.

**Comment by tylerl0706** [comment]
@rkeithhill off the top of your head, how might one test for this? I can try to set up a VM to test.

**Comment by rkeithhill** [comment]
You can try with a VM that has access to the Windows ISO image for installation but has **never** had an internet connection.  In our case, it was folks running our code on submarines and in air-gapped satellite test systems.  In talking to the engineer who has run into this issue over the years, he says it can also happen on a system that has been connected to the internet **if** there is a problem with the root certificate on that system.

**Comment by rkeithhill** [comment]
@AmandaCameron Good point - should have read your original post.  On Linux, I don't believe PowerShell Core does anything with script signatures.  The typical set of AuthenticodeSignature commands are not available on that platform.  Same behavior with the latest update - 1.8.2?

**Comment by AmandaCameron** [comment]
Just tested, and indeed it's still an issue under 1.8.2 -- worth noting I have my login shell also set to powershell (`pwsh-preview` to be exact) and it loads up fine,  both in my normal terminal as well as in the VSCode embedded terminal ( **NOT** The Powershell Integrated Console, though )

**Comment by andyleejordan** [comment]
The main scenario here — the extension freezing for minutes while scaffolding with Plaster — no longer applies: **Plaster integration was removed in v2024.2.1**, and the execution architecture has been rewritten since 2018 (PSReadLine now runs in-process). If you still hit startup/IntelliSense slowness offline on a current build, please open a fresh issue with logs. Closing as obsolete.

<sub>🤖 Drafted by GitHub Copilot at @andyleejordan's request — pending maintainer review.</sub>
